Daemon/Daemon/XPCListener.m

```diff
@@ -204,7 +204,8 @@ -(BOOL)listener:(NSXPCListener *)listener shouldAcceptNewConnection:(NSXPCConnec
     
     //step 2: validate
     // check that client is signed with Objective-See's and it's BlockBlock
-    if(errSecSuccess != (status = SecTaskValidateForRequirement(taskRef, (__bridge CFStringRef)(requirement))))
+    OSStatus validateForRequirementStatus = SecTaskValidateForRequirement(taskRef, (__bridge CFStringRef)(requirement));
```

Comment:
I have ran static analyzer, and it showed that the assignment of the `SecTaskValidateForRequirement` was not being evaluated in the condition, thus most likely it was not behaving correctly.

For your consideration. Might work, might not. 🤞🏻

Reply:
not that the analyzer is wrong, but we can just re-use `status` here like so:

```
status = SecTaskValidateForRequirement(taskRef, (__bridge CFStringRef)(requirement));
if(errSecSuccess != status) {
   goto bail;
}
 ```
 
Could always put in some logging those error cases if we wanted to print out why it failed (e.g. what `status` is)

Reply:
Thank you! I have missed the fact that this is the last check for `status` variable. In general I refuse to reuse variables. They are optimized out anyway, so it is a matter of avoiding bugs in future 👍🏻